js/api.js

```diff
@@ -7,17 +7,32 @@ const warn = (message) => {
     console.warn(message)
   }
 }
-const urlParams = (() => {
-  return queryString.parse(window.location.search)
-})()
+const urlParam = (name) => {
+  return queryString.parse(window.location.search)[name]
+}
```

Comment:
With this change the URL params are parsed on each call rather than once and cached. In some apps we remove URL params like short-lived tokens during the running of the app. If that's a possibility here then parsing the URL once and caching the results may be preferable.

Reply:
Good point, thanks. Although I've changed it so this module can be tested (tests are also part of this PR). If it's cached in the module level (like it used to be before), once you import a module in your test suite, you can't do anything. At this point it doesn't seem likely we're going to hide any of these params, so I'd rather keep it testable for now. Not sure if there's some easy solution that would handle both cases well.

Reply:
What I've done in several projects is have `urlParams` in its own module, so that it can be imported/required by any module that needs it. For testing purposes it could then be mocked, but that may be more effort than it's worth at this point.